### PR TITLE
[Request][Port] How to prepare, cook and serve an Ethereal Crystal

### PR DESCRIPTION
**Pull request by Bobbanz1**
## About The Pull Request
Completes the request posted by Seryn aka Light Bulb Main Extraordinaire aka **_CHIEF LIGHT BULB_** through a Discord Direct Message:

Screenshot 2023-01-04 210635

Ports the following PRs from TG:
* https://github.com/tgstation/tgstation/pull/57225
* https://github.com/tgstation/tgstation/pull/60613
* https://github.com/tgstation/tgstation/pull/57434
* https://github.com/tgstation/tgstation/pull/58650
* https://github.com/tgstation/tgstation/pull/59068 (Something...don't remember what)
* https://github.com/tgstation/tgstation/pull/67102
* https://github.com/tgstation/tgstation/pull/68184
* https://github.com/tgstation/tgstation/pull/68868
* https://github.com/tgstation/tgstation/pull/66853

General description of what this PR does for those who don't want to crawl through those PRs to find out themselves:

This PR basically adds a resurrection mechanism to Ethereals alongside a draw-back to this resurrection mechanism, through this PR they gain an automated resurrection method in the form of crystalization, although at the draw-back of gaining a permanent brain trauma in exchange, said trauma can be anywhere between mild to severe and stacks with each death.

Another draw-back to this is that there is a cooldown, so if killed within this time-span after having already gone through the Crystalization, then they are literally just dead and have to be revived as usual by medbay.

The revival crystal can also be shattered by someone beating at it sufficiently long enough, which will shatter the crystal and terminate the revival process, leaving the Ethereal as a dead body which has to be revived by medbay. 

They can be disarmed before they turn into a crystal, to delay the crystalizing for 2 minutes. (This can be used to perform surgery on the ethereal before it crystalizes), if you attack them while they are crystalizing it will cause the crystalizing to completely stop after a while.

## Why It's Good For The Game
Differentiates Ethereals from other species by giving them the Crystalization mechanic from tgstation rather than having them be yet another re-skinned human with a few changes to them.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://user-images.githubusercontent.com/59128051/210644252-df80e917-724e-4e3a-8812-cd089ad3b1da.mp4

</details>

## Changelog
:cl:Bobbanz1, ghost, Qustinnus, Kryson, uglykoala, KittyNoodle, itseasytosee, FlamingLily, SmArtKar, Fikou, ShizCalev, wesoda25
add: Ethereals now crystalize on death, allowing them to heal themselves unless the crystal is destroyed, but giving them a brain trauma when its succesful.
fix: ethereal hair will now change with ethereal color without an external update
fix: If you get qdeleted while crystalizing as ethereal, you no longer get super-owned
fix: Ethereals will no longer crystallize after being eaten. (His grace, etc, etc)
fix: Fixed ethereal revive crystal bugging out when buckled
fix: ethereals no longer revive after being changeling drained
qol: ethereals can now press the ghost button to cancel crystallization and permanently vacate their body
/:cl:

### CODE REVIEW

**Comment by Karmic-Skink** [comment]
As this is a large change to how a species operates, could you please first port it to Bee.

If they accept it, we can action it from there.

**Comment by Bobbanz1** [comment]
> As this is a large change to how a species operates, could you please first port it to Bee.
> 
> If they accept it, we can action it from there.

I’ll see what I can do tomorrow

**Comment by Bobbanz1** [comment]
Reopening this PR due to the Lightbulb asking me to do so

File: code/modules/surgery/organs/heart.dm [hunk 1]
```diff
@@ -231,3 +231,229 @@
 		owner.heal_overall_damage(15, 15, 0, BODYTYPE_ORGANIC)
 		if(owner.reagents.get_reagent_amount(/datum/reagent/medicine/ephedrine) < 20)
 			owner.reagents.add_reagent(/datum/reagent/medicine/ephedrine, 10)
+
+
+//NSV13 - Rest of file is purely ours from down here - Ethereal Crystalization
+
```
Comment (covertcorvid) [inline code/modules/surgery/organs/heart.dm:237]: Code looks alright. I would prefer this section in a separate file in the NSV folder.

**Review by covertcorvid — APPROVED** [review]

**Pull request merged**